internal/saucecloud/imagerunner.go

```diff
@@ -313,6 +317,57 @@ func (r *ImgRunner) PollRun(ctx context.Context, id string, lastStatus string) (
 	}
 }
 
+func extractFile(artifactFolder string, file *zip.File) error {
+	fullPath := path.Join(artifactFolder, file.Name)
+
+	relPath, err := filepath.Rel(artifactFolder, fullPath)
+	if err != nil {
+		return err
+	}
+	if strings.Contains(relPath, "..") {
+		return fmt.Errorf("file %s is relative to an outside folder", file.Name)
+	}
+
+	folder := path.Dir(fullPath)
+	if err := os.MkdirAll(folder, 0755); err != nil {
+		return err
+	}
+
+	fd, err := os.Create(fullPath)
+	if err != nil {
+		return err
+	}
+	defer fd.Close()
+
+	rd, err := file.Open()
+	if err != nil {
+		return err
+	}
+	defer rd.Close()
+
+	_, err = io.Copy(fd, rd)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func saveToTempFile(closer io.ReadCloser) (string, error) {
+	fd, err := os.CreateTemp("", "")
+	if err != nil {
+		return "", err
+	}
```

Comment:
```suggestion
	}
	defer fd.Close()
	defer closer.Close()
```

Resource leaks otherwise.

Reply:
I kept the `fd.Close()` to the end of the function. We are still interested by a safe Close() as it will be the input for later in the process.